scripts/add_electricity.py

```diff
@@ -457,32 +457,33 @@ def attach_hydro(n, costs, ppl):
             missing_plants = pd.Index(inflow_buses.unique()).difference(
                 inflow.indexes["plant"]
             )
-            intersection_plants = pd.Index(
+            plants_with_data = pd.Index(
                 inflow_buses[inflow_buses.isin(inflow.indexes["plant"])]
             )
 
             # if missing time series are found, notify the user and exclude missing hydro plants
             if not missing_plants.empty:
                 # original total p_nom
                 total_p_nom = ror.p_nom.sum() + hydro.p_nom.sum()
-                idxs_to_keep = inflow_buses[
-                    inflow_buses.isin(intersection_plants)
-                ].index
+                idxs_to_keep = inflow_buses[inflow_buses.isin(plants_with_data)].index
```

Comment:
I think we need to improve the readability of the code because it is easy to get lost with indices.
Something that Irealize now is that this code seems a duplication and we may need to debug it together.

Initially is defined as the values of a subset of inflow_buses, whose index are idxs_to_keep.
I think that to improve the readability, we may avoid defining idx_to_keep inside the if but outside.
plants with data may be the view: `inflow_buses[inflow_buses.isin(inflow.indexes["plant"])]` where its index is idxs_to_leep and its values are '''plants'''.
We may revise the code in that way so that it may be easier to understand.

What do you think? Let me know if the above is clear

Reply:
@davide-f agree that this change look very natural. However, I needed about an hour to understand it, so I absolutely aligned with your idea to make this fragment a bit more readable! :D

Have implemented a possible approach. Along with revision of `idxs_to_keep` definition, there were following changes:
1) move reading PHS `phs = ppl.query('technology == "Pumped Storage"')` to a less over-crowded part;
2) revise the names of the data frames `ror` and `hydro` to make it clear that both were derived from `ppl` and contain run-of-river and reservoir data, respectively;
3) change `inflow_idx` to `hydro_ppl_idx` to avoid an association with `inflow` indices which can be a bit misleading.

Not sure if all these changes are really necessary. I'd suggest to select only those which are really needed. Could you please check if these changes do really allow to decrease reading time? :)